url80587fc7d5_dennervidal_CloudSim_DVFS_tgz-pJ8-Examples_Dvfs_example_simpleJ8/src/org/cloudbus/cloudsim/UtilizationModelPlanetLabInMemory.java

```diff
@@ -0,0 +1,74 @@
+package org.cloudbus.cloudsim;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+
+/**
+ * The Class UtilizationModelPlanetLab.
+ */
+public class UtilizationModelPlanetLabInMemory implements UtilizationModel {
+
+	/** The scheduling interval. */
+	private double schedulingInterval;
+
+	/** The data (5 min * 288 = 24 hours). */
+	private final double[] data = new double[289];
+
+	/**
+	 * Instantiates a new utilization model PlanetLab.
+	 * 
+	 * @param inputPath the input path
+	 * @throws NumberFormatException the number format exception
+	 * @throws IOException Signals that an I/O exception has occurred.
+	 */
+	public UtilizationModelPlanetLabInMemory(String inputPath, double schedulingInterval)
+			throws NumberFormatException,
+			IOException {
+		setSchedulingInterval(schedulingInterval);
+		BufferedReader input = new BufferedReader(new FileReader(inputPath));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *input*. The resource is referenced at lines 29, 32. The resource is closed at line 35. There are other execution paths that do not contain closure statements, for example, when BufferedReader.readLine throws an exception. Ensure *input* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).